### PR TITLE
TL Mbox read and write for stm32wb

### PR DESCRIPTION
**Pull request by OueslatiGhaith**
Hello,

This pull request is related to #1397 and #1401, inspired by #24, built upon the work done in #1405 and #1424, and was tested on an stm32wb55rg.

This pull request aims to add read and write functionality to the TL mailbox for stm32wb microcontrollers

### CODE REVIEW

**Comment by Dirbaio** [comment]
What happened with the tests in #1424 @OueslatiGhaith @xoviat? I see there's a test there but it's not enabled. I'd like to have at least basic "it works" tests, so we don't break it in the future.

If I flashed the radio stack wrong, or it needs another version, please let me know and I'll fix it!

File: examples/stm32wb/src/bin/tl_mbox_tx_rx.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            code,
+            payload_len,
+            payload[3..]
+        );
+    }
+
+    mbox.shci_ble_init(&mut ipcc, Default::default());
+
+    info!("resetting BLE");
+    mbox.send_ble_cmd(&mut ipcc, &[0x01, 0x03, 0x0c]);
```
Comment (xoviat) [inline examples/stm32wb/src/bin/tl_mbox_tx_rx.rs:78]: When running on hw, the example stalls here. Is this desired behavior?

**Comment by xoviat** [comment]
TODO (for me): adjust the linker file for the tests for wb55

File: examples/stm32wb/src/bin/tl_mbox_tx_rx.rs (hunk 1 above)
Comment (OueslatiGhaith) [reply examples/stm32wb/src/bin/tl_mbox_tx_rx.rs:78]: no, should reset the ble stack and return a command complete packet, I'll look into it

Comment (OueslatiGhaith) [reply examples/stm32wb/src/bin/tl_mbox_tx_rx.rs]: it was an incorrect payload, the cmd should be [0x01, 0x03, 0x0c, 0x00, 0x00]

**Comment by xoviat** [comment]
That part works now. There are some issues, but they can be addressed in the next pr. I think HIL will pass.

bors r+

**Pull request merged**